packages/frontend-2/components/project/page/settings/general/block/ProjectInfo.vue

```diff
@@ -0,0 +1,160 @@
+        class="mb-4"
+        :disabled="disabled"
+      />
+      <FormTextArea
+        v-model="localProjectDescription"
+        name="projectDescription"
+        label="Project Description"
+        placeholder="Description (optional)"
+        show-label
+        color="foundation"
+        :disabled="disabled"
+      />
+      <template #bottom-buttons>
+        <FormButton text :disabled="!hasChanges" @click="resetLocalState">
+          Cancel
+        </FormButton>
+        <FormButton :disabled="!hasChanges" @click="emitUpdate">Update</FormButton>
+      </template>
+    </ProjectPageSettingsBlock>
+
+    <LayoutDialog
+      v-model:open="showConfirmDialog"
+      max-width="md"
+      :buttons="dialogButtons"
+    >
+      <template #header>Unsaved Changes</template>
+      <div class="space-y-4">
+        <p>You have unsaved changes. Do you want to save them before leaving?</p>
+      </div>
+    </LayoutDialog>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { onBeforeRouteLeave, useRouter } from 'vue-router'
+import { PencilIcon } from '@heroicons/vue/24/outline'
+import {
+  FormTextInput,
+  FormTextArea,
+  FormButton,
+  LayoutDialog
+} from '@speckle/ui-components'
+import type { ProjectPageSettingsGeneralBlockProjectInfo_ProjectFragment } from '~~/lib/common/generated/gql/graphql'
+import type { RouteLocationRaw } from 'vue-router'
+import { graphql } from '~~/lib/common/generated/gql'
+
+graphql(`
+  fragment ProjectPageSettingsGeneralBlockProjectInfo_Project on Project {
+    id
+    role
+    name
+    description
+  }
+`)
+
+const props = defineProps<{
+  project: ProjectPageSettingsGeneralBlockProjectInfo_ProjectFragment
+  disabled?: boolean
+}>()
+
+const emit = defineEmits(['update-project'])
+const router = useRouter()
+
+const targetRoute = ref<RouteLocationRaw | undefined>(undefined)
+const localProjectName = ref(props.project.name)
+const localProjectDescription = ref(props.project.description ?? '')
+const showConfirmDialog = ref(false)
+
+const hasChanges = computed(() => {
+  return (
+    localProjectName.value !== props.project.name ||
+    localProjectDescription.value !== props.project.description
+  )
+})
+
+const emitUpdate = () => {
+  emit('update-project', {
+    name: localProjectName.value,
+    description: localProjectDescription.value,
+    onComplete: handleRedirection
+  })
+}
+
+const handleRedirection = () => {
+  if (targetRoute.value) {
+    router.push(targetRoute.value)
+    targetRoute.value = undefined
+  }
+}
+
+const resetLocalState = () => {
+  localProjectName.value = props.project.name
+  localProjectDescription.value = props.project.description ?? ''
+  showConfirmDialog.value = false
+}
+
+const dialogButtons = computed(() => [
+  {
+    text: 'Discard Changes',
+    props: { color: 'secondary', fullWidth: true, outline: true },
+    onClick: () => {
+      showConfirmDialog.value = false
+      resetLocalState()
+      if (targetRoute.value) {
```

Comment:
1. Duplicates implementation of handleRedirection
2. 2. Doesn't handle targetRoute cleanup